### PR TITLE
Add support for Deno language server

### PR DESCRIPTION
**Pull request by mkhattab**
This PR address #2235 and adds support for the Deno language server.

### CODE REVIEW

File: clients/lsp-javascript.el [hunk 1]
```diff
@@ -206,5 +206,101 @@ particular FILE-NAME and MODE."
...
+  "Initialization options for the Deno language server."
+  `(:enable t
+    :config ,lsp-clients-deno-config
+    :importMap ,lsp-clients-deno-import-map
+    :lint ,(lsp-json-bool lsp-clients-deno-enable-lint)
+    :unstable ,(lsp-json-bool lsp-clients-deno-enable-unstable)
+    :codeLens (:implementations ,(lsp-json-bool lsp-clients-deno-enable-code-lens-implementations)
+               :references ,(lsp-json-bool (or lsp-clients-deno-enable-code-lens-references
+                                               lsp-clients-deno-enable-code-lens-references-all-functions))
+               :referncesAllFunctions ,(lsp-json-bool lsp-clients-deno-enable-code-lens-references-all-functions))))
```
Comment (yyoncho) [inline clients/lsp-javascript.el:293]: seems like here you(or the server) have a typo: referncesAllFunctions

**Review by yyoncho — COMMENTED** [review]
Looks good - 2 minor issues and it is good to go. Can you add `:package-version '(lsp-mode . "7.1.0")` to all `defcustoms`?

File: clients/lsp-javascript.el (hunk 1 above)
Comment (mkhattab) [reply clients/lsp-javascript.el]: It's a typo on my end. Good catch 👍

**Review by yyoncho — APPROVED** [review]

**Comment by yyoncho** [comment]
Thank you for contributing to `lsp-mode`!

**Pull request merged**